pages/api/signup.ts

```diff
@@ -6,18 +6,18 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     }
     try {
         let {firstName, lastName, email, laptop, experience, grade, shirt, diet, other} = req.body;
-        let errors = []; //grouping: names, email, long paragraphs, button choices
-        const names: string[] = [firstName, lastName];
+        let errors = [];
+        const names: string[] = [firstName, lastName, email];
         const explains: string[] = [experience, diet, other];
         const choices: any[] = [laptop, grade, shirt];
-        const combinedArr = names.concat(explains.concat(choices));
+        const combinedArr = names.concat(email.concat(choices));
         const emailPattern = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
         const namePattern = /^[a-zA-Z\s'-]{2,25}$/;
-        const explainPattern = /^[a-zA-Z0-9\s.,'!?\-()]{2,500}$/;
+        const explainPattern = /^[a-zA-Z0-9\s.,'!?\-()]{0,500}$/;
 
         for (let i = 0; i < combinedArr.length; i++) {
             if (combinedArr[i] == null) {
-                return res.status(400).json({ message: "Empty Input"});
+                return res.status(400).json({ message: "Empty Required Inputs"});
             }
         }
 
```

Comment:
Nice work. The only issues I saw is that when the email isn't put, it gives the email error (which is ok) + the name error (which isn't supposed to show).

The other issue is that if both name fields, for example, are blank, it gives two name errors. I think it will work if you filter it out in the loop saying that if it's already there, don't add another.